src/main.cpp

```diff
@@ -1,617 +1,41 @@
-        continue;
-      } else if (setupResult == SetupInstanceResults::SelectAnother) {
-        InstanceManager::singleton().clearCurrentInstance();
-        return RestartExitCode;
-      } else {
-        return 1;
-      }
-    }
-
-    checkPathsForSanity(*currentInstance.gamePlugin(), settings);
-
-    organizer.setManagedGame(currentInstance.gamePlugin());
-    organizer.createDefaultProfile();
-
-    log::info(
-      "using game plugin '{}' ('{}', variant {}, steam id '{}') at {}",
-      currentInstance.gamePlugin()->gameName(),
-      currentInstance.gamePlugin()->gameShortName(),
-      (settings.game().edition().value_or("").isEmpty() ?
-        "(none)" : *settings.game().edition()),
-      currentInstance.gamePlugin()->steamAPPId(),
-      currentInstance.gamePlugin()->gameDirectory().absolutePath());
-
-
-    CategoryFactory::instance().loadCategories();
-    organizer.updateExecutablesList();
-    organizer.updateModInfoFromDisc();
-
-    organizer.setCurrentProfile(currentInstance.profileName());
-
-    if (auto r=handleCommandLine(cl, organizer)) {
-      return *r;
-    }
-
-    auto splash = createSplash(settings, dataPath, currentInstance.gamePlugin());
-
-    QString apiKey;
-    if (GlobalSettings::nexusApiKey(apiKey)) {
-      ni.getAccessManager()->apiCheck(apiKey);
-    }
-
-    log::debug("initializing tutorials");
-    TutorialManager::init(
-        qApp->applicationDirPath() + "/"
-            + QString::fromStdWString(AppConfig::tutorialsPath()) + "/",
-        &organizer);
-
-    if (!application.setStyleFile(settings.interface().styleName().value_or(""))) {
-      // disable invalid stylesheet
-      settings.interface().setStyleName("");
-    }
-
-    int res = 1;
-
-    { // scope to control lifetime of mainwindow
-      // set up main window and its data structures
-      MainWindow mainWindow(settings, organizer, *pluginContainer);
-
-      ni.getAccessManager()->setTopLevelWidget(&mainWindow);
-
-      QObject::connect(&mainWindow, SIGNAL(styleChanged(QString)), &application,
-                       SLOT(setStyleFile(QString)));
-      QObject::connect(&instance, SIGNAL(messageSent(QString)), &organizer,
-                       SLOT(externalMessage(QString)));
-
-      log::debug("displaying main window");
-      mainWindow.show();
-      mainWindow.activateWindow();
-
-      if (splash) {
-        // don't pass mainwindow as it just waits half a second for it
-        // instead of proceding
-        splash->finish(nullptr);
-      }
-
-      tt.stop();
-
-      res = application.exec();
-      mainWindow.close();
+  initLogging();
 
-      ni.getAccessManager()->setTopLevelWidget(nullptr);
-    }
+  QApplication::setAttribute(Qt::AA_EnableHighDpiScaling);
+  MOApplication app(cl, argc, argv);
 
-    settings.geometry().resetIfNeeded();
-    return res;
-  }
-  catch (const std::exception &e)
-  {
-    reportError(e.what());
+  SingleInstance instance(cl.multiple());
```

Comment:
So here with instance you mean a process right?
Maybe for the sake of clearness you could use "process" as the variable name here? Or maybe "processInstance"

Reply:
It's actually named `singleInstance` everywhere in `MOApplication` to try to avoid confusion. I'd like to to rename it or just get rid of it entirely, probably integrate it in `MOApplication`, it's a 100 lines.

Reply:
I've also run into the same problem for the command line help, but I just spelled it out instead: https://github.com/ModOrganizer2/modorganizer/blob/346a6babdf6263568b7fe20b9019d3f4ca0d68f4/src/commandline.cpp#L374.

Reply:
Renamed to `MOMultiProcess`.